python/desc/twinkles/twinkles_sky.py

```diff
@@ -0,0 +1,160 @@
+    """
+    def __init__(self,
+                 obs_metadata,
+                 brightestStar_gmag_inCat=11.0,
+                 brightestGal_gmag_inCat=11.0,
+                 availableConnections=None,
+                 sntable='TwinkSN',
+                 sn_sedfile_prefix='spectra_files/specFile_'):
+        """
+        Parameters
+        ----------
+        obs_metadata : instance of `lsst.sims.utils.ObservationMetaData`
+            Observational MetaData associated with an OpSim Pointing
+        brightestStar_gmag_inCat : optional, float, defaults to 11.0
+            brightest star allowed in terms of its g band magnitude in
+            'ab' magsys. The reason for this constraint is to keep phosim
+            catalog generation times in check.
+        brightestGal_gmag_inCat : optional, float, defaults to 11.0
+            brightest galaxy allowed in terms of its g band magnitude in
+            'ab' magsys. The reason for this constraint is to keep phosim
+            catalog generation times in check.
+        availableConnections : list of connections, optional, defaults to None
+            list of available connections for DBObject
+        sntable : string, optional, defaults to `TwinkSN`
+            Name of the table on fatboy with the SN parameters desired. 
+        sn_sedfile_prefix : string, optional, defaults to `spectra_files/specFile_'
+            prefix for sed of the supernovae.
+        Attributes
+        ----------
+        snObj : CatalogDBObj for SN
+        available_connections : available connections
+
+        ..notes : 
+        """
+        # Observation MetaData
+        self.obs_metadata = obs_metadata
+
+        # Constraint on the brightest star: 
+        self.brightestStar_gmag_inCat = brightestStar_gmag_inCat
+        self.brightestStarMag = 'gmag > {}'.format(self.brightestStar_gmag_inCat)
+
+        # Constraint on brightest galaxy
+        self.brightestGal_gmag_inCat = brightestGal_gmag_inCat
+        self.brightestGalMag = 'g_ab > {}'.format(self.brightestGal_gmag_inCat)
+
+        self.availableConnections = availableConnections
+
+        # Lists of component phosim Instance Catalogs and CatalogDBObjects
+        # Stars
+        self.compoundStarDBList = [StarObj, CepheidStarObj]
+        self.compoundStarICList = [PhoSimCatalogPoint, PhoSimCatalogPoint]
+
+        # Galaxies
+        self.compoundGalDBList = [GalaxyBulgeObj, GalaxyDiskObj, GalaxyAgnObj]
+        self.compoundGalICList = [PhoSimCatalogSersic2D, PhoSimCatalogSersic2D,
+                                  TwinklesCatalogZPoint]
+
+        # SN 
+        ## SN catalogDBObject
+        if self.availableConnections is None:
+            self.availableConnections = list()
+            self.snObj = SNDBObj(table=sntable, connection=None)
+            self.availableConnections.append(self.snObj.connection)
+        else:
+            self.snObj = SNDBObj(table=sntable, connection=self.availableConnections[0])
```

Comment:
This is only valid if the `self.availableConnections[0]` is a connection to the database that `self.snObj` expects.  It will not be generalizable for users who want to use their own database (but maybe that is not something we want to support).